tests/testthat/test-plot-construction.R

```diff
@@ -0,0 +1,16 @@
+context("plot-construction")
+
+test_that("Custom object cannot be added without generic", {
+  p <- ggplot()
+  custom_object <- structure(list(), class = 'test_object')
+  expect_error(p + custom_object, "Don't know how to add custom_object to a plot")
+})
+
+ggplot_add.test_object <- function(object, p, objectname) {
+  10
+}
+test_that("Methods can be defined for adding custom objects", {
```

Comment:
You can define the method inside the test, just to make things slightly cleaner.

Reply:
For some reason I’m having problems getting dispatch to work correctly in the test. Moving it out fixed it locally but it still throws an error on travis..?

Reply:
Do you have any idea why methods defined inside test_that expressions are not being dispatched to?

Reply:
Yes, it's something to do with the weird way in which S3 dispatch happens. I forget all the details, but just move it outside for now.